### PR TITLE
Add binary logs and publish logger artifacts

### PR DESCRIPTION
**Pull request by zippy1981**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [*] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [*] Build system
 

### Purpose
This does two things immediately:

* Creates a binary log of the C# build that can be among other things read with http://msbuildlog.com/.
* Exposes the current text msbuild logs and new binary as appbeyor artifacts.

Long term, I'm working on [a project that will parse msbuild binary logs](https://github.com/justaprogrammer/MSBuildLogOctokitChecker/graphs/contributors) and report data on a pr page with my friends @StanleyGoldman and @profet23.

So at some point I will send a PR to have AppVeyor also push this log to the website and  then more info will be appear on the PR "checks page."

This will list warnings and stylecop type things for the C# code.

### CODE REVIEW

File: bin/projects/dbatools/msbuild.rsp [hunk 1]
```diff
@@ -6,4 +6,5 @@
 /ConsoleLoggerParameters:NoSummary;Verbosity=quiet
 # Log to a file and set options
 /filelogger
-/fileloggerparameters:Verbosity=detailed
\ No newline at end of file
+/fileloggerparameters:Verbosity=detailed
+/logger:BinaryLogger,".\bin\StructuredLogger.dll";".\msbuild.binlog"
```
Comment (zippy1981) [inline bin/projects/dbatools/msbuild.rsp]: Fun facts here:

* The file name needs to end in `.binlog`. 
* Relative paths need to be prepended to `.\` to be parsed.

File: bin/build-project.ps1 [hunk 2]
```diff
@@ -1,8 +1,9 @@
 [CmdletBinding()]
```
Comment (zippy1981) [inline bin/build-project.ps1]: I'm actually not sure how to manually trigger this build so I didn't test this.

The appveyor tests don't trigger a build this way. Maybe we should fix that.

File: bin/build-project.ps1 [hunk 3]
```diff
@@ -34,22 +35,23 @@ if (-not (Test-Path $msbuild)) {
...
+$MsbuildOptions = $MsbuildOptions + '/logger:BinaryLogger,"{0}";"{1}" ' -f (Join-Path $PSScriptRoot 'StructuredLogger.dll'), (Resolve-Path '..\msbuild.bin.log').Path  
+
 if ( -not (Test-Path $ProjectPath)) {
     throw new-object 'System.IO.FileNotFoundException' 'Could not file project or solution', $ProjectPath
 }
 
-Write-Verbose -Message "Building the library with command $msbuild $ProjectPath /p:Configuration=$msbuildConfiguration $msbuildOptions /t:$MsBuildTarget"
-& $msbuild $ProjectPath "/p:Configuration=$msbuildConfiguration" $msbuildOptions "/t:$MsBuildTarget"
+Write-Verbose -Message "Building the library with command $msbuild $ProjectPath /p:Configuration=$msbuildConfiguration $MsbuildOptions /t:$MsBuildTarget"
+& $msbuild $ProjectPath "/p:Configuration=$msbuildConfiguration" $MsbuildOptions "/t:$MsBuildTarget"
```
Comment (StanleyGoldman) [inline bin/build-project.ps1]: You aren't capturing msbuild errors, if the build fails, you will not be sending MSBLOC your failed build log.

https://github.com/justaprogrammer/MSBuildLogOctokitChecker/blob/master/docs/integrating.md#integrating-on-appveyor

Comment (zippy1981) [reply bin/build-project.ps1]: This is a good point, I think I need to use `on_failure:` to artefacts of a failed build.

**Comment by potatoqualitee** [comment]
all checks pass now - this is out of my league, let me know when it's good to merge and thank you all! 💯

File: bin/projects/dbatools/msbuild.rsp (hunk 1 above)
Comment (potatoqualitee) [reply bin/projects/dbatools/msbuild.rsp]: hey zips, how we lookin as far as progress and merging?

**Comment by wsmelton** [comment]
@potatoqualitee just reference if you needed/wanted to know what is being implemented here is for the strict purpose of adding artifacts back to Appveyor:
image

So we can get better insight into the build of the library itself and be able to troubleshoot things when it fails in Appveyor-world. At a future date the parsing that @zippy1981 is referencing will offer the ability to have that more easily accessible in a PR and provide the direct details without having to look over the logs manually....If I understand everything in this PR at least.

**Comment by wsmelton** [comment]
You can go back and look at failed PRs and see that artifacts tab in Appveyor has nothing in it.

Which overall it might be something we further along by having the dbatools logs captured as well for a given session...then publish that as an artifact too. Having the logs accessible as artifacts will make troubleshooting issues with test running in Appveyor much easier.

**Comment by potatoqualitee** [comment]
ohh, that actually sounds useful, thank you @wsmelton 👍

**Comment by potatoqualitee** [comment]
zips, your PR misses you

**Comment by potatoqualitee** [comment]
hey zip, please reopen PR once you are ready to proceed. it's not cancelled, just postponed ;)

**Comment by StanleyGoldman** [comment]
Apologies, we are wrapping up a release of the service soon.
Hopefully next week you guys can be our first guinea pigs.

**Comment by StanleyGoldman** [comment]
For more information on what this actually is: https://msblocweb.azurewebsites.net/

**Comment by zippy1981** [comment]
@cl Please merge this as is. The bin logs as artefacts are useful in and of themselves. I'll make another PR with further changes.

File: bin/build-project.ps1 (hunk 3 above)
Comment (wsmelton) [reply bin/build-project.ps1]: @zippy1981 are you going to make this change?

**Comment by potatoqualitee** [comment]
will let zipps submit the requested change in later PR 👍

**Comment by potatoqualitee** [comment]
Also, thanks for the info, @StanleyGoldman ! that looks really cool

**Pull request merged**